### PR TITLE
Do not trim off the end of NuGet package names

### PR DESCRIPTION
**Pull request by JoeRobich**
Package names are being trimmed at the last `.` in the Remove Unused References dialog.

image

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]

File: src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs [hunk 1]
```diff
@@ -124,7 +124,9 @@ public bool TryGetValue(string keyName, out object? content)
                             break;
                         case UnusedReferencesTableKeyNames.ReferenceName:
                             // It is unnecessary to display the full path to project and assembly files.
-                            content = Path.GetFileNameWithoutExtension(ReferenceUpdate.ReferenceInfo.ItemSpecification);
+                            content = ReferenceUpdate.ReferenceInfo.ReferenceType == ReferenceType.Project
+                                ? Path.GetFileNameWithoutExtension(ReferenceUpdate.ReferenceInfo.ItemSpecification)
+                                : ReferenceUpdate.ReferenceInfo.ItemSpecification;
```
Comment (ryzngard) [inline src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs:129]: The comment and the code are out of sync here. You're only trimming for project references, not assembly files

**Review by ryzngard — APPROVED** [review]

File: src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs [hunk 2]
```diff
@@ -124,7 +124,9 @@ public bool TryGetValue(string keyName, out object? content)
                             break;
                         case UnusedReferencesTableKeyNames.ReferenceName:
                             // It is unnecessary to display the full path to project and assembly files.
-                            content = Path.GetFileNameWithoutExtension(ReferenceUpdate.ReferenceInfo.ItemSpecification);
+                            content = ReferenceUpdate.ReferenceInfo.ReferenceType == ReferenceType.Project
+                                ? Path.GetFileNameWithoutExtension(ReferenceUpdate.ReferenceInfo.ItemSpecification)
```
Comment (davidwengier) [inline src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs:128]: Is it even trimming off anything here? Is it valid to have both `Foo.csproj` and `Foo.vbproj` as project references?

Comment (JoeRobich) [reply src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs:128]: Yeah. Let's just ditch it for consistency. Was trying to mimic the Solution Explorer and let the icon do the talking. =)

File: src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs (hunk 1 above)
Comment (JoeRobich) [reply src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs:129]: Think I got it sorted now

**Comment by JoeRobich** [comment]
@vatsalyaagrawal For 16.11 Preview 3

**Pull request merged**